### DISCUSSION TITLE
Flutter web support

### DISCUSSION THREAD

**Discussion by Padi142** [mongo-dart/mongo_dart#347] [General]
When trying to acces the databse via `Db.create()` on Flutter web, I get this error:

```
Error: MongoDB ConnectionException: Could not connect to monkidatabse-shard-00-00.cxodm.mongodb.net:27017
- Unsupported operation: default SecurityContext getter
    at Object.throw_ [as throw] (http://localhost:58926/dart_sdk.js:5080:11)
at mongo_dart._ConnectionManager.new.open (http://localhost:58926/packages/mongo_dart/src/database/utils/split_hosts.dart.lib.js:11102:23)
    at open.throw (<anonymous>)
    at http://localhost:58926/dart_sdk.js:40646:38
    at _RootZone.runBinary (http://localhost:58926/dart_sdk.js:40515:59)
    at _FutureListener.thenAwait.handleError (http://localhost:58926/dart_sdk.js:35449:33)


```
It would be very nice to be able to use Mongo db on Flutter web.

**Comment by giorgiofran** [comment]
Unfotunately it is not so easy to solve this issue. I cannot promise anything.

**Comment by prl900** [comment]
I just found this and was wondering if there are any updates. @giorgiofran, if you can describe the problem a little bit I could try to help with finding a solution. Thanks

**Comment by giorgiofran** [comment]
The problem is that web apps cannot use `dart:io`. I have found a package that seems to make sockets usable also on web apps, but I have to test it and I will do it in the new version of mongo_dart that I'm developing.

**Comment by prl900** [comment]
Thanks for the update @giorgiofran. Looking forward to seeing that new implementation, it sounds very promising.

**Comment by giorgiofran** [comment]
The goal is to implement also transactions, but it is a big step forward, and probably it will not be compatible with the present package.

**Comment by TheMettaCode** [comment]
I'm a bit new at this and it's probably a bit of a simple question but I've been able to interact with my database via Atlas and their REST API option at https://www.mongodb.com/docs/atlas/api/ for development. Problem is, risk of API exposure in production. I know it's tricky but if there is a way of obscuring api keys it may be an option.

So my question would be do you think its possible to integrate something like that into the package?

**Comment by giorgiofran** [comment]
It is possible, the problem is to find the time to develop it. At present I'm updating and renewing the driver, so I have no time to give a look into this, sorry.
